src/app/store/trips/selectors.ts

```diff
@@ -0,0 +1,69 @@
+import { createFeatureSelector, createSelector } from '@ngrx/store';
+import { Trip } from '../../models/trips';
+import { TripState, tripsFeatureKey } from './reducer';
+
+export const selectFeature = createFeatureSelector<TripState>(tripsFeatureKey);
+
+const deriveDatesAndCost = (trip: Trip) => {
+  let startDate = new Date(8640000000000000);
+  let endDate = new Date(-8640000000000000);
+  let totalTripCost = 0;
+
+  if (trip.itinerary && trip.itinerary.length > 0) {
+    trip.itinerary.forEach(activity => {
+      const newStartDate = activity.startDate.toDate();
+      const newEndDate = activity.endDate.toDate();
+
+      startDate = startDate
+        ? newStartDate < startDate
+          ? newStartDate
+          : startDate
+        : newStartDate;
+      endDate = endDate
+        ? newEndDate > endDate
+          ? newEndDate
+          : endDate
+        : newEndDate;
+      totalTripCost += activity.cost || 0;
+    });
+  }
+
+  if (
+    startDate === new Date(8640000000000000) ||
+    endDate === new Date(-8640000000000000)
```

Comment:
you could maybe have the initial dates something like `undefined` or `null`, instead of an arbitrary date that doesn't make sense

Reply:
For reasons unknown, the undefined or null was not working so that is why I opted for the funny numbers

Reply:
Typescript nonsense loses the type later on because of filter/map/reduce typescript bugs that are fixed in the upcoming release of TS